test/docfx.Test/e2e/E2ETest.cs

```diff
@@ -154,21 +164,27 @@ private static void DeleteDirectory(string targetDir)
             Directory.Delete(targetDir, false);
         }
 
-        private static IEnumerable<string> FindTestSpecHeadersInFile(string path)
+        private static IEnumerable<(string, bool only)> FindTestSpecHeadersInFile(string path)
         {
-            var sections = File.ReadAllText(path).Split("\n---", StringSplitOptions.RemoveEmptyEntries);
+            var content = File.ReadAllText(path);
+            var sections = content.Split("\n---", StringSplitOptions.RemoveEmptyEntries);
 
             foreach (var section in sections)
             {
+                var only = false;
                 var yaml = section.Trim('\r', '\n', '-');
                 var header = YamlUtility.ReadHeader(yaml) ?? "";
-
+#if DEBUG
+                if (header.Contains("ONLY"))
```

Comment:
What if "only" is needed in the header text? Can we make it more strict as in the beginning, using `header.StartsWith("[only]", StringComparison.OrdinalIgnoreCase)`?

Reply:
we already have that case. check # Orphan file(not explicitly referenced in toc file)'s toc, only look up parent folder toc files